Fix layout offset on physical Android devices after closing feedback

go-run-jump opened the pull request:
## Summary
This PR fixes a layout offset issue on physical Android devices where the main app content remains offset to the right/down after closing the feedback view.

## The Issue
On physical Android devices running Flutter 3.24+, when users open and then close the feedback view (either by submitting feedback or cancelling), the underlying application screen content remains offset. This issue does not occur on Android emulators.

## The Fix
The fix is simple - explicitly position the screenshot widget at `Offset.zero` when feedback is not displayed in the `_FeedbackLayoutDelegate.performLayout` method.

```dart
if (\!displayFeedback) {
  layoutChild(_screenshotId, BoxConstraints.tight(size));
  // Explicitly position at zero offset - fixes layout offset on physical Android devices
  positionChild(_screenshotId, Offset.zero);
  return;
}
```

## Testing
This fix has been tested on a physical Android device and resolves the offset issue.

## Related Issues
This addresses the issue reported in #322

## Type of Change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

go-run-jump committed:
Fix layout offset on physical Android devices after closing feedback

When closing the feedback view on physical Android devices, the main app content
remains offset to the right/down. This fix explicitly positions the screenshot
widget at Offset.zero when feedback is not displayed.

This addresses the issue reported in #322 where the layout remains offset after
interacting with the feedback view on physical Android devices.

Potentially related to #322

ueman commented:
> This fix has been tested on a physical Android device and resolves the offset issue.

Can you please post your testing steps? also, can you try writing a widget test, please?

go-run-jump committed:
test: add regression test for layout positioning

Add widget test to ensure screenshot positioning behavior is maintained.
The test verifies that the screenshot widget is positioned at (0,0) when
feedback is closed, both initially and after closing feedback.

Note: This is a regression test that passes with or without the fix,
as the actual layout offset issue only occurs on physical Android devices
and cannot be reproduced in the test environment.

JakesMD commented:
This fixed the issue in our web app. Thanks @go-run-jump! I've switch to go-run-jump's branch for now. Hope we can get this through soon.

Niburume commented:
worked for me on IOS. Thanks to @go-run-jump

famasf1 commented:
This worked. It should've been merge since last year. What the heck.
